### PR TITLE
joystickLocation determined by joystick.properties

### PR DESCRIPTION
**Pull request by 4thwind**

### CODE REVIEW

File: src/org/usfirst/frc/team2175/properties/JoystickProperties.java [hunk 1]
```diff
@@ -27,13 +28,20 @@ protected void populate() {
...
+        gearIntakeOutInfo = getInfo("button.gearintake.out");
+        runShooterInInfo = getInfo("button.shooter.in");
+        runFeederInInfo = getInfo("button.feeder.in");
+    }
+
+    public String[] getInfo(String str) {
+        String[] data = getStringPropertyValue(str).split(",");
+        data[1] = data[1].trim();
+        return data;
     }
```
Comment (mabster314) [inline src/org/usfirst/frc/team2175/properties/JoystickProperties.java]: Do you have tests written for these? If not, it would probably be a good idea

Comment (4thwind) [inline src/org/usfirst/frc/team2175/properties/JoystickProperties.java]: Tests for what?
When testing it, I made sure it returned an integer, if that is what you are talking about..

File: src/org/usfirst/frc/team2175/properties/JoystickProperties.java [hunk 2]
```diff
@@ -54,30 +62,58 @@ public double getDeadbandValue() {
...
-    public int getRunShooterIn() {
-        return runshooterin;
+    public String getRunShooterInLocation() {
+        return runShooterInInfo[0];
     }
 
-    public int getRunFeederIn() {
-        return runfeederin;
+    public String getRunFeederInLocation() {
+        return runFeederInInfo[0];
```
Comment (bvisness) [inline src/org/usfirst/frc/team2175/properties/JoystickProperties.java]: Doing things this way adds a lot of methods....but we can maybe shrink this a lot using inner classes.

Here's a small example: https://gist.github.com/bvisness/51281c09ca03bd4b1b9c5012bea78ace

File: src/org/usfirst/frc/team2175/driverstation/DriverStation.java [hunk 3]
```diff
@@ -29,24 +29,50 @@ public DriverStation() {
...
+        gearIntakeOutButton = new JoystickButton(
+                location(joystickProperties.getGearIntakeOutLocation()),
                 joystickProperties.getGearIntakeOutNumber());
 
         deadbandSize = joystickProperties.getDeadbandValue();
         deadbandCalculator = new DeadbandCalculator();
         ServiceLocator.register(this);
     }
 
+    private Joystick location(String str) {
```
Comment (bvisness) [inline src/org/usfirst/frc/team2175/driverstation/DriverStation.java]: `location` is not a very good name for this. It really returns a `Joystick`, so maybe something like `joystickForName(String name)` would be better?

**Pull request merged**